sklearnex/_utils.py

```diff
@@ -32,14 +32,22 @@ def set_sklearn_ex_verbose():
                       '"WARNING", "INFO", "DEBUG".'.format(logLevel))
 
 
-def get_patch_message(s):
+def get_patch_message(s, queue=None):
     import sys
     if s == "onedal":
         message = "running accelerated version on "
-        if 'daal4py.oneapi' in sys.modules:
+        if queue is not None:
+            if queue.sycl_device.is_gpu:
+                message += 'GPU'
+            elif queue.sycl_device.is_cpu:
+                message += 'CPU'
+            else:
+                raise RuntimeError('Unsupported device')

```

Comment:
is this an unreachable branch of code?

Reply:
No - for example, we do not support FPGA